File: services/Services.BackupCreator/Jobs/CreateBackupJob.cs
using Dropbox.Api;
using Dropbox.Api.Files;
using Microsoft.Extensions.Logging;
using Quartz;
using Services.BackupCreator.Config;
using System;
using System.Collections.Generic;
using System.IO;
using System.IO.Compression;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace Services.BackupCreator.Jobs
{
    public class CreateBackupJob : IJob
    {
        private readonly ILogger _logger;
        private readonly ZipBackupCreator _backupCreator;
        private readonly DropboxSender _dropboxSender;

        public CreateBackupJob(ILogger<BackupCreatorService> logger, ZipBackupCreator backupCreator,
            DropboxSender dropboxSender)
        {
            _logger = logger;
            _backupCreator = backupCreator;
            _dropboxSender = dropboxSender;
        }

        public async Task Execute(IJobExecutionContext context)
        {
            _logger.LogInformation("Backup job started");
            try
            {
                var backupFile = _backupCreator.CreateBackup();
                await _dropboxSender.BackupToDropbox(backupFile);
            }
            catch (Exception e)
            {
                _logger.LogWarning(e, "Error in backup creation");
            }
        }
    }
}

File: services/SiteParser/create_backup_job/create_backup_job.py
from .zip_backup_creator import ZipBackupCreator
from .dropbox_sender import DropboxSender


class CreateBackupJob:
    def __init__(self, backup_creator: ZipBackupCreator, dropbox_sender: DropboxSender):
        self.__backup_creator = backup_creator
        self.__dropbox_sender = dropbox_sender

    def execute(self):
        print("Backup job started")
        try:
            backup_path = self.__backup_creator.create_backup()
            self.__dropbox_sender.backup_to_dropbox(backup_path)
        except Exception as error:
            print("Error in backup creation: " + str(error))
